core/src/Measurement.hpp

```diff
@@ -1,96 +1,126 @@
+    Eigen::Vector3d com_linear_acceleration{Eigen::Vector3d::Zero()};              ///< Center of mass (COM) linear acceleration
+    Eigen::Matrix3d position_slip_cov{Eigen::Matrix3d::Identity()};                ///< Covariance of position slip
+    Eigen::Matrix3d orientation_slip_cov{Eigen::Matrix3d::Identity()};             ///< Covariance of orientation slip
+    Eigen::Matrix3d position_cov{Eigen::Matrix3d::Identity()};                     ///< Covariance of position
+    Eigen::Matrix3d orientation_cov{Eigen::Matrix3d::Identity()};                  ///< Covariance of orientation
+    Eigen::Matrix3d com_position_process_cov{Eigen::Matrix3d::Identity()};         ///< Covariance of COM position process noise
+    Eigen::Matrix3d com_linear_velocity_process_cov{Eigen::Matrix3d::Identity()};  ///< Covariance of COM linear velocity process noise
+    Eigen::Matrix3d external_forces_process_cov{Eigen::Matrix3d::Identity()};       ///< Covariance of external forces process noise
+    Eigen::Matrix3d com_position_cov{Eigen::Matrix3d::Identity()};                 ///< Covariance of COM position
+    Eigen::Matrix3d com_linear_acceleration_cov{Eigen::Matrix3d::Identity()};       ///< Covariance of COM linear acceleration
 };
 
+/**
+ * @struct OdometryMeasurement
+ * @brief Represents odometry measurements including base position, orientation, and their covariances
+ */
 struct OdometryMeasurement {
-    double timestamp{};
-    Eigen::Vector3d base_position{Eigen::Vector3d::Zero()};
-    Eigen::Quaterniond base_orientation{Eigen::Quaterniond::Identity()};
-    Eigen::Matrix3d base_position_cov{Eigen::Matrix3d::Identity()};
-    Eigen::Matrix3d base_orientation_cov{Eigen::Matrix3d::Identity()};
+    double timestamp{};                                 ///< Timestamp of the measurement
+    Eigen::Vector3d base_position{Eigen::Vector3d::Zero()};      ///< Base position from odometry
+    Eigen::Quaterniond base_orientation{Eigen::Quaterniond::Identity()}; ///< Base orientation from odometry
+    Eigen::Matrix3d base_position_cov{Eigen::Matrix3d::Identity()};      ///< Covariance matrix of base position
+    Eigen::Matrix3d base_orientation_cov{Eigen::Matrix3d::Identity()};   ///< Covariance matrix of base orientation
 };
 
+/**
+ * @struct TerrainMeasurement
+ * @brief Represents terrain height measurements at a specific timestamp
+ */
 struct TerrainMeasurement {
-    double timestamp{};
-    double height{};
-    double height_cov{1.0};
-    TerrainMeasurement(double timestamp, double height, double height_cov)
-        :timestamp(timestamp), height(height), height_cov(height_cov){}
+    double timestamp{};                                 ///< Timestamp of the measurement
+    double height{};                                    ///< Terrain height measurement
+    double height_cov{1.0};                             ///< Covariance of terrain height measurement (default 1.0)
```

Comment:
"Covariance measures the direction of the relationship between two variables" so double height_cov{1.0}?

Reply:
yep formally it's a variance but we could say it's a covariance if it measures the relationship to itself e.g. cov_xx. Just keeping it for naming consistency across all measurements.